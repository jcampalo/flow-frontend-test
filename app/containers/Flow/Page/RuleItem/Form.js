import styled from 'styled-components';

export default styled.form`
  background: #FFF;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  overflow: hidden;
  height: 100%;
  padding: 2em;
`;
