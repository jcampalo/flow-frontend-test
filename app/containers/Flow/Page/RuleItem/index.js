import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import TextArea from 'components/TextArea';
import Input from 'components/Input';
import messages from '../../messages';
import Form from './Form';
import Wrapper from './Wrapper';
import InputWrapper from './InputWrapper';

class RuleItem extends PureComponent {
  handleChange = (value, name) => {
    const { index, onChange } = this.props;

    onChange({ name, value, index });
  }

  render() {
    const { id = '', title = '', body = '', rulePass = '', ruleFail = '' } = this.props;

    return (
      <Wrapper>
        <Form>
          <InputWrapper>
            <label htmlFor="title">
              <FormattedMessage {...messages.inputTitle}>
                {placeholder => (
                  <Input
                    name="title"
                    type="text"
                    placeholder={placeholder}
                    value={title}
                    onChange={this.handleChange}
                  />
                )}
              </FormattedMessage>
            </label>
          </InputWrapper>
          <InputWrapper>
            <label htmlFor="id">
              <FormattedMessage {...messages.inputId}>
                {placeholder => (
                  <Input
                    name="id"
                    type="number"
                    placeholder={placeholder}
                    value={id}
                    onChange={this.handleChange}
                  />
                )}
              </FormattedMessage>
            </label>
          </InputWrapper>
          <InputWrapper>
            <label htmlFor="body">
              <FormattedMessage {...messages.inputBody}>
                {placeholder => (
                  <TextArea
                    rows={6}
                    name="body"
                    type="text"
                    placeholder={placeholder}
                    value={body}
                    onChange={this.handleChange}
                  />
                )}
              </FormattedMessage>
            </label>
          </InputWrapper>
          <InputWrapper>
            <label htmlFor="rulePass">
              <FormattedMessage {...messages.inputRulePass}>
                {placeholder => (
                  <Input
                    name="rulePass"
                    type="number"
                    placeholder={placeholder}
                    value={rulePass}
                    onChange={this.handleChange}
                  />
                )}
              </FormattedMessage>
            </label>
          </InputWrapper>
          <InputWrapper>
            <label htmlFor="ruleFail">
              <FormattedMessage {...messages.inputRuleFail}>
                {placeholder => (
                  <Input
                    name="ruleFail"
                    type="number"
                    placeholder={placeholder}
                    value={ruleFail}
                    onChange={this.handleChange}
                  />
                )}
              </FormattedMessage>
            </label>
          </InputWrapper>
        </Form>
      </Wrapper>
    );
  }
}

RuleItem.propTypes = {
  id: PropTypes.string,
  index: PropTypes.number,
  title: PropTypes.string,
  body: PropTypes.string,
  rulePass: PropTypes.string,
  ruleFail: PropTypes.string,
  onChange: PropTypes.func
};

export default RuleItem;
