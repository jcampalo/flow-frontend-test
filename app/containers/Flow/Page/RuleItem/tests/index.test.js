import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { IntlProvider } from 'react-intl';

import RuleItem from '../index';

const initial = {};
const renderComponent = props => {
  const finalProps = { ...initial, ...props };

  return (
    <IntlProvider {...props}>
      <RuleItem {...finalProps} />
    </IntlProvider>
  );
};

describe('<RuleItem />', () => {
  it('should render component ', () => {
    const tree = create(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call onChange when input changes', () => {
    const onChange = jest.fn();
    const name = 'name';
    const value = 'value';
    const index = 'index';
    const props = {
      index,
      onChange
    };
    const component = shallow(<RuleItem {...props} />);
    const instance = component.instance();

    instance.handleChange(value, name);

    expect(onChange).toHaveBeenCalledWith({ index, name, value });
  });
});
