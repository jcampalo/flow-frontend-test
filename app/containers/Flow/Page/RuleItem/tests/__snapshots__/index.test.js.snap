// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RuleItem /> should render component  1`] = `
<div
  className="Wrapper-hPVDJR hjkowz"
>
  <form
    className="Form-gTeswK dyrSex"
  >
    <div
      className="InputWrapper-bpDXXy bZGuau"
    >
      <label
        htmlFor="title"
      >
        <span
          className="Wrapper-ddcghJ kYIiWq"
        >
          <label
            className="Label-fIbAhr kVyEfu"
          >
            Title
          </label>
          <input
            autoComplete="off"
            className="StyledInput__Input-eYzgai zNgEU"
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </span>
      </label>
    </div>
    <div
      className="InputWrapper-bpDXXy bZGuau"
    >
      <label
        htmlFor="id"
      >
        <span
          className="Wrapper-ddcghJ kYIiWq"
        >
          <label
            className="Label-fIbAhr kVyEfu"
          >
            Id
          </label>
          <input
            autoComplete="off"
            className="StyledInput__Input-eYzgai zNgEU"
            name="id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="number"
            value=""
          />
        </span>
      </label>
    </div>
    <div
      className="InputWrapper-bpDXXy bZGuau"
    >
      <label
        htmlFor="body"
      >
        <span
          className="Wrapper-dyuznp gAhBoi"
        >
          <label
            className="Label-ghyrbC KoSFt"
          >
            Body
          </label>
          <textarea
            autoComplete="off"
            className="StyledTextArea__TextArea-cqmHOB jGwssY"
            name="body"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            rows={6}
            type="text"
            value=""
          />
        </span>
      </label>
    </div>
    <div
      className="InputWrapper-bpDXXy bZGuau"
    >
      <label
        htmlFor="rulePass"
      >
        <span
          className="Wrapper-ddcghJ kYIiWq"
        >
          <label
            className="Label-fIbAhr kVyEfu"
          >
            If rule pased
          </label>
          <input
            autoComplete="off"
            className="StyledInput__Input-eYzgai zNgEU"
            name="rulePass"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="number"
            value=""
          />
        </span>
      </label>
    </div>
    <div
      className="InputWrapper-bpDXXy bZGuau"
    >
      <label
        htmlFor="ruleFail"
      >
        <span
          className="Wrapper-ddcghJ kYIiWq"
        >
          <label
            className="Label-fIbAhr kVyEfu"
          >
            If rule failed
          </label>
          <input
            autoComplete="off"
            className="StyledInput__Input-eYzgai zNgEU"
            name="ruleFail"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="number"
            value=""
          />
        </span>
      </label>
    </div>
  </form>
</div>
`;
